### ISSUE TITLE
Reverse Proxy Problem

### ISSUE THREAD

**Issue by beefycheddar** [tidusjar/Ombi#2364]
#### Ombi build Version:

V 3.0.3421

#### Update Branch:

Open Beta

#### Media Sever:

Plex

#### Operating System:

Windows 10

#### Problem Description:

Maybe I didn't understand the directions.
Using IIS. I have the web.config sample text configured with my domain and placed in the wwwroot\ombi directory, however IIS keeps serving the default IIS page. 

I removed "ombi/" from the end of the rewrite line so it looks like this: 
`<action type="Rewrite" value="http{R:1}://my.domain.com/{R:2}" />`

When I visit my.domain.com, I get the IIS page. When I visit my.domain.com/ombi, I get a 500 error. 
If I add "ombi/" back to the rewrite line, the results are the same.

Any ideas on what I've done wrong?

Thanks

**Comment by dxtrlws** [comment]
I’ll post what I have for my web.config file tomorrow, but you can turn on detailed error messages in IIS to get a better idea on what’s wrong. Did you set the http variables in IIS?

**Comment by dxtrlws** [comment]
Here's a copy of my config file. I'm running over SSL. 
```
<?xml version="1.0" encoding="UTF-8"?>
<configuration>
    <system.webServer>

        <defaultDocument>
            <files>
            </files>
        </defaultDocument>

        <rewrite>
            <rules>
                <clear />
                <rule name="ReverseProxyInboundOMBI" stopProcessing="true">
                    <match url="(.*)" />
                    <action type="Rewrite" url="http://localhost:5000/ombi/{R:1}" />
                    <serverVariables>
                        <set name="host" value="$host" />
                        <set name="HTTP_X_FORWARDED_HOST" value="$host" />
                        <set name="HTTP_X_REAL_IP" value="$remote_addr" />
                        <set name="HTTP_X_FORWARDED_FOR" value="$proxy_add_x_forwarded_for" />
                        <set name="HTTP_X_FORWARDED_PROTO" value="$scheme" />
                        <set name="HTTP_X_FORWARDED_SSL" value="on" />
                    </serverVariables>
                    <conditions trackAllCaptures="true">
                    </conditions>
                </rule>
            </rules>

            <outboundRules>
                <clear />

                <rule name="Restore Encoding" preCondition="Restore HTTP_ACCEPT_ENCODING}" enabled="true">
                    <match serverVariable="HTTP_ACCEPT_ENCODING" pattern="^(.+)" />
                    <conditions logicalGrouping="MatchAll" trackAllCaptures="true" />
                    <action type="Rewrite" value="{HTTP_X_ORIGINAL_ACCEPT_ENCODING}" />
                </rule>

                <rule name="ReverseProxyOutboundOMBI" preCondition="ResponseIsHtml1" enabled="true" stopProcessing="false">
                    <match filterByTags="A, Area, Base, Form, Frame, Head, IFrame, Img, Input, Link, Script" pattern="^http(s)?://localhost:5000/ombi/(.*)" />
                    <conditions logicalGrouping="MatchAll" trackAllCaptures="true">
                        <add input="{HTTP_REFERER}" pattern="/ombi/" />
                    </conditions>
                    <action type="Rewrite" value="http{R:1}://mydomain/ombi/{R:2}" />
                </rule>

		<preConditions>
			<preCondition name="ResponseIsHtml">
				<add input="{RESPONSE_CONTENT_TYPE}" pattern="^text/html" />
			</preCondition>
			<preCondition name="Restore HTTP_ACCEPT_ENCODING}">
                        	<add input="{RESPONSE_CONTENT_TYPE}" pattern=".+" />
			</preCondition>
		</preConditions>

            </outboundRules>
        </rewrite>
        <security>
            <authentication>
            </authentication>
        </security>
        <urlCompression doStaticCompression="true" doDynamicCompression="false" />
    </system.webServer>
</configuration>

```

**Comment by PowerShellCrack** [comment]
I am having the same issue. I have ombi extract to c:\ombi. I have it running as a service and I can access via localhost:5000 url. I'm also running IIS 10 (On windows 10). I have ARR installed and configured as a reverse proxy.  I have a custom Web Site setup going to my homepage (www.trickysplexondemand.com), and on that website folder, I add a custom Application (\ombi)pointing to c:\inetpub\wwroot\ombi and using DefaultAppPool. I copied the web.config into that folder and changed the mydomain to "http{R:1}://www.trickysplexondemand.com/ombi/{R:2}". When I navigate to http://www.trickysplexondemand.com/ombi I get a "500 - Internal server error." The reverse proxy is working because I have subsonic running under tomcat and I can access it through iis with the rewrite going to port 8080. If I remove the servervariables and restoring encoding rules, I get a Loading... page. It just stays there. I made a copy of my webconfig here: www.trickysplexondemand.com/ombi.web.config.txt

**Comment by PowerShellCrack** [comment]
oh and my ombi configuration base url is to /ombi. And I restated the service after the change

**Comment by PowerShellCrack** [comment]
I was able to resolve it. At the server level you must add these to approved server variables:
host
HTTP_HOST
HTTP_X_FORWARDED_HOST
HTTP_X_REAL_IP
HTTP_X_FORWARDED_FOR
HTTP_X_FORWARDED_PROTO
HTTP_X_FORWARDED_SSL
HTTP_ACCEPT_ENCODING
HTTP_X_ORIGINAL_ACCEPT_ENCODING
RESPONSE_CONTENT_TYPE

I found an article that explained it: https://weblogs.asp.net/owscott/500-50-error-using-url-rewrite

**Comment by PowerShellCrack** [comment]
well heck, I just found out outside my network it still comes back as localhost:5000. any ideas?

**Comment by anojht** [comment]
@PowerShellCrack could you try removing baseurl inside Ombi and restarting the service. Does this work?

**Comment by PowerShellCrack** [comment]
I got it to work now. I found there was a little more to it. This web config has the serverVariables. IIS 10 wasn't excepting them until I added them to the root. Also I had to make sure I configured OMBI correctly. The baseURL original wasn't set but then I noticed the web.config has a it going there so I configured it; next if the Application URL wasn't configure so when you use plex to sign it, it redirects to here: https://plex.tv/auth#?forwardUrl=http://localhost:5000/ombi/login....Thanks
